### PR TITLE
simplify storage SIGINT signal handler

### PR DESCRIPTION
**Pull request by liuyu85cn**
#### What type of PR is this?
- [X] bug
- [ ] feature
- [ ] enhancement

#### What does this PR do?

In short: 

1. let's say we(storage server) have a thread running some task.

2. user send kill SIGINT signal, linux schedule the signal handler at the thread we talked about.

3. now, we interrupt the previous task, executing the signal handler, 
    it calls storage server->stop()
    trying to clear everything storage server has(also include this thread!!!!). 

4. Then we stuck.

#### Which issue(s)/PR(s) this PR relates to?

[Storage hang at exit #3434](https://github.com/vesoft-inc/nebula/issues/3434)

### CODE REVIEW

File: src/daemons/StorageDaemon.cpp [hunk 1]
```diff
@@ -180,7 +180,7 @@ void signalHandler(int sig) {
     case SIGTERM:
       FLOG_INFO("Signal %d(%s) received, stopping this server", sig, ::strsignal(sig));
       if (gStorageServer) {
-        gStorageServer->stop();
+        gStorageServer->notifyStop();
```
Comment (kikimo) [inline src/daemons/StorageDaemon.cpp:183]: This fix looks good. My concern, don't forget to fix metad too!

**Review by yixinglu — APPROVED** [review]
LGTM

**Review by kikimo — APPROVED** [review]
LGTM

**Review by critical27 — APPROVED** [review]
Good job

**Pull request merged**